src/map/utils/battleutils.cpp

```diff
@@ -6769,7 +6769,7 @@ namespace battleutils
 
         float sumFastCast = std::clamp<float>((float)(fastCast + uncappedFastCast + inspirationFastCast), -100.f, 100.f);
 
-        return (uint32)(cast * ((100.0f - sumFastCast) / 100.0f));
+        return std::clamp(static_cast<int32>((cast * ((100.0f - sumFastCast) / 100.0f))), 1000, std::numeric_limits<int32>::max());
```

Comment:
I am confused on why we are now casting this to an int?
I would try this and see if this still gives you an issue.
`return std::max(static_cast<uint32>(cast * ((100.0f - sumFastCast) / 100.0f)), 1000);`

Reply:
I'll give it a go but the reason I did it because things like cast time for summons are flat reductions so you can in theory go negative. In My testing I made sure to go negative and it would read the value as max automatically. From my understanding uint32 can't handle negatives. I'm happy to be wrong but that's why.

Reply:
`std::max` should just pick the maximum value. If its below 1000 it will automatically default to 1000

Reply:
ok so tried what you suggested with a minor adjustment to clear an error. ```return std::max(static_cast<uint32>(cast * ((100.0f - sumFastCast) / 100.0f)), static_cast<uint32>(1000));``` I'm basically back to an infinite  cast time. when the value goes negative the cast time becomes infinite. uint32 cannot handle negative values so when you give it a negative value it goes to the largest bit value an works in reverse. Example: ```return std::max(static_cast<uint32>(cast * ((100.0f - sumFastCast) / 100.0f)), static_cast<uint32>(1000)); == return std::max(static_cast<uint32>(-1), static_cast<uint32>(1000)); == return 4294967295;``` the returned value is not seen below 1000 it is register well above and we will be waiting 150~ years for that cast to complete.